code/datums/antagonists/datum_traitor.dm

```diff
@@ -84,15 +84,15 @@
 /datum/antagonist/traitor/apply_innate_effects()
 	if(owner.assigned_role == "Clown")
 		var/mob/living/carbon/human/traitor_mob = owner.current
-		if(traitor_mob&&istype(traitor_mob))
-			if(!silent) 
+		if(traitor_mob && istype(traitor_mob))
+			if(!silent)
 				to_chat(traitor_mob, "Your training has allowed you to overcome your clownish nature, allowing you to wield weapons without harming yourself.")
 			traitor_mob.dna.remove_mutation(CLOWNMUT)
 
 /datum/antagonist/traitor/remove_innate_effects()
 	if(owner.assigned_role == "Clown")
 		var/mob/living/carbon/human/traitor_mob = owner.current
-		if(traitor_mob&&istype(traitor_mob))
+		if(traitor_mob && istype(traitor_mob) && traitor_mob.mind.special_role != "traitor")
```

Comment:
I think all antags should be exempt from this madness. Just checking for the presence of `special_role` will do it I think.

Reply:
Other roles have their own ways of fixing this, and theirs still work I think. It's just traitor that doesn't. I'll check into this

Reply:
FYI, this was fixed in an antag refactor on tg: https://github.com/tgstation/tgstation/pull/34236